src/Motor.Extensions.Hosting.Kafka/KafkaMessageConsumer.cs

```diff
@@ -76,9 +76,13 @@ public async Task ExecuteAsync(CancellationToken token = default)
                 try
                 {
                     var msg = _consumer?.Consume(token);
-                    if (msg != null && !msg.IsPartitionEOF)
+                    if (msg is { IsPartitionEOF: false })
                     {
-                        SingleMessageHandling(token, msg);
+                        Task<bool> completed;
```

Comment:
(nitpick) Why is `completed` a `Task<bool>` and not just a `bool`? Is there any reason against `await`-ing directly in the `do` loop?

Reply:
Then `Consume_SynchronousMessageHandling_DoesNotProcessSecondMessageBeforeFirstMessageIsSuccessful` fails because we don't start processing multiple messages anymore. That's actually what I thought would happen when I replied to @pkemkes' comment earlier. However, the critical await is this one, while @pkemkes' comment was legitimate.

Reply:
I see, this is basically the critical point of this PR: You really only want to `await` if `SynchronousMessageHandling == true`.